examples/springboot/spring-boot_app/src/main/java/com/canonical/sampleapp/service/S3Service.java

```diff
@@ -0,0 +1,78 @@
+/*
+* Copyright 2025 Canonical Ltd.
+* See LICENSE file for licensing details.
+*/
+
+package com.canonical.sampleapp.service;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Component;
+
+import software.amazon.awssdk.core.exception.SdkServiceException;
+import software.amazon.awssdk.services.s3.S3Client;
+
+@ConditionalOnProperty(name = "spring.cloud.aws.credentials.accessKey")
+@Component
+public class S3Service {
+    private final S3Client s3Client;
+    private final ResourceLoader resourceLoader;
+
+    S3Service(S3Client s3Client, ResourceLoader resourceLoader) {
+        this.s3Client = s3Client;
+        this.resourceLoader = resourceLoader;
+    }
+
+    public Resource getObject(String bucketName, String key) throws IOException {
+        final Resource resource = resourceLoader.getResource(String.format("s3://%s/%s", bucketName, key));
```

Comment:
You can extract `String.format("s3://%s/%s", bucketName, key)`  to a variable to avoid defining it twice